src/core/plugins/app-manager.ts

```diff
@@ -797,6 +808,17 @@ export default class AppManager extends CallablePlugin {
     }
   }
 
+  /**
+   @return {AppContext[]} - returns all contexts that include the input node.
+   */
+  getNodeAllContexts(node: MuonNodeInfo): AppContext[] {
+    const currentTime = getTimestamp()
+    return Object.values(this.appContexts)
+      .filter((ctx:AppContext) => {
+        return (!ctx.expiration || ctx.expiration > currentTime) && ctx.party.partners.includes(node.id)
```

Comment:
Is sending non-expired contexts enough?
Sometimes the nodes need to have the last 2 context for an app.